internal/services/appservice/static_web_app_function_app_registration_resource.go

```diff
@@ -0,0 +1,191 @@
+package appservice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/web/2023-01-01/staticsites"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type StaticWebAppFunctionAppRegistrationResource struct{}
+
+var _ sdk.Resource = StaticWebAppFunctionAppRegistrationResource{}
+
+type StaticWebAppFunctionAppRegistrationModel struct {
+	StaticWebAppID string `tfschema:"static_web_app_id"`
+	FunctionAppID  string `tfschema:"function_app_id"`
+}
+
+func (r StaticWebAppFunctionAppRegistrationResource) ResourceType() string {
+	return "azurerm_static_web_app_function_app_registration"
```

Comment:
worth calling this `association` rather than `registration` for consistency within the provider?

Reply:
Given the service nomenclature, and the fact it updates the target Function App authentication, I think registration is probably appropriate here?